### ISSUE TITLE
Local best-checkpoint monitors hard-coded val/ACC, not the model_selector key_metric (val/AUC_ROC) — local/federated best-model mismatch

### ISSUE THREAD

**Issue by Ultimate-Storm** [KatherLab/MediSwarm#409]
## Problem

The **local** "best checkpoint" and the **federated** "best global model" are selected on **different metrics**, so for every AUROC-objective job the locally-saved `best_*.pt` is chosen on the *wrong* metric.

- **Local (per-client, per-epoch)** — Lightning `ModelCheckpoint` monitors a **hard-coded `val/ACC`** (accuracy), `mode="max"`:
  ```python
  # application/jobs/_shared/custom/threedcnn_ptl.py:533
  to_monitor = "val/ACC"
  min_max = "max"
  checkpointing = ModelCheckpoint(monitor=to_monitor, mode=min_max, save_top_k=1, save_last=True)
  ```
  `to_monitor` is a literal — it is never read from env/config/`key_metric` (only occurrence is line 533).

- **Federated (per-round, across the swarm)** — NVFlare `IntimeModelSelector` uses **`key_metric = "val/AUC_ROC"`** (all challenge jobs + `ODELIA_ternary_classification`; `config_fed_client.conf`).

So "best local epoch" (highest accuracy) and "best global round" (highest AUROC) optimize different objectives.

## Why this looks half-finished

`application/jobs/_shared/custom/models/base_model.py:174-178` logs **both** `val/ACC` and `val/AUC_ROC`, and explicitly re-logs AUROC with the comment **"For ModelCheckpoint, also log as 'val/AUC_ROC'"** — i.e. AUROC was exposed *for the checkpoint callback*, but `ModelCheckpoint.monitor` was never switched from `val/ACC` (or made `key_metric`-driven). This is the local counterpart of the earlier fix that set the federated selector's `key_metric` to `val/AUC_ROC`.

## Fix

Drive the local `ModelCheckpoint` from the **same `key_metric`** the federated selector uses, so local and federated agree:

- [ ] Plumb `key_metric` (from `config_fed_client.conf`'s `model_selector` args) to the training code — e.g. surface it as an env var alongside the existing `odelia_*` env vars (`env_config.py` / `env_vars`), or read it from config in the executor.
- [ ] In `threedcnn_ptl.py:533`, set `to_monitor = <key_metric>` (default `val/ACC` for back-compat) instead of the literal.
- [ ] Derive `mode` from the metric direction (max for `val/AUC_ROC` / `val/ACC`, min for `val/loss`) — mirror the `IntimeModelSelector` higher-is-better/`negate_key_metric` convention so the two never disagree.
- [ ] Confirm the chosen metric is logged under exactly that name at epoch end (`val/AUC_ROC` already is; `val/ACC`, `val/loss` too).
- [ ] Keep `save_last=True` so warm-continue's `mediswarm_latest_global.pt` mirror path is unaffected.

## Verification

- [ ] Unit/local: run a short local training with `key_metric=val/AUC_ROC`; confirm the retained `best_*.pt` corresponds to the epoch with the **highest AUROC**, not the highest accuracy.
- [ ] Confirm `best_FL_global_model.pt` / the challenge-eval path and the federated `IntimeModelSelector` now agree on the metric.
- [ ] Default (`key_metric` unset) still monitors `val/ACC` (no behavior change for jobs that want accuracy, e.g. `minimal_training_pytorch_cnn`).

## Refs
- `application/jobs/_shared/custom/threedcnn_ptl.py:533-542` (hard-coded `val/ACC`)
- `application/jobs/_shared/custom/models/base_model.py:174-178` (both metrics logged; the "for ModelCheckpoint" comment)
- `application/jobs/ODELIA_ternary_classification/app/config/config_fed_client.conf:162` (`key_metric = "val/AUC_ROC"`)
